Fix some things I broke and simplify

jjcastro opened the pull request.

benwulfe reviewed (changes requested):
please put a description of the PR for what was fixed and what you did to test (specific manual steps)

jjcastro commented:
Right:

To prevent the function from triggering on loops: I thought the `firestore_payload.old_value` and `firestore_payload.value` were dicts but they were protos, so this filter wasn't catching anything. So now I converted the proto to dict, and turns out that comes with a  updateMask so we can just substract the ouput fields and check if any input fields are modified.

And then, unrelated, I had also messed up some imports, and put the EE authentication in the main.py so that it doesn't depend on one of them authenticating (in case one section fails).

To test, I put the contents of this code by modifying the cloud function directly and modified a sample via firebase, and saw it ran as expected: https://screenshot.googleplex.com/7JARtaKyLy9Z9bw

Then I checked the logs of the function and see it's running as expected, catching the exception for the known issue with "oxygen" and so on (https://screenshot.googleplex.com/689WX9XbvgrjwEq) and also skipping the "only outputs modified" trigger

benwulfe reviewed (approved):
thank you for the description!
